File: matrix/masolas/masolas.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<double> Masolas(const vector<double>& lista)
{
    vector<double> result(lista.size());
    for (int i = 0; i < lista.size(); i++)
    {
        result[i] = lista[i] * lista[i];
    }
    return result;
}

int main()
{
    vector<double> v = { 4, 5, 1, 2, 3 };
    for (double& elem : Masolas(v))
    {
        cout << elem << " ";
    }
}

File: matrix/masolas/masolas.py
def masolas(lista):
    result = []
    for elem in lista:
        result.append(elem*elem)
    return result

l = [4,5,1,3,2]

print(masolas(l))
# bár ezt pythonban list comprehension-nel szoktuk, ami valószínűleg gyorsabb is, mert kizárt, hogy resize-ol.
print([x*x for x in l])
